sympy/physics/continuum_mechanics/beam.py

```diff
@@ -0,0 +1,567 @@
+    >>> b.deflection()
+    (-7*x + SingularityFunction(x, 0, 3)/2 - SingularityFunction(x, 2, 4)/4 + 3*SingularityFunction(x, 4, 3)/2)/(E*I)
+    """
+
+    def __init__(self, length, elastic_modulus, second_moment):
+        self._length = length
+        self._elastic_modulus = elastic_modulus
+        self._second_moment = second_moment
+        self._boundary_conditions = {'deflection': [], 'moment': [], 'slope': []}
+        self._load = 0
+
+    def __str__(self):
+        str_sol = 'Beam(%s, %s, %s)' % (sstr(self._length), sstr(self._elastic_modulus), sstr(self._second_moment))
+        return str_sol
+
+    __repr__ = __str__
+
+    @property
+    def length(self):
+        """Length of the Beam."""
+        return self._length
+
+    @length.setter
+    def length(self, l):
+        self._length = sympify(l)
+
+    @property
+    def elastic_modulus(self):
+        """Young's Modulus of the Beam. """
+        return self._elastic_modulus
+
+    @elastic_modulus.setter
+    def elastic_modulus(self, e):
+        self._elastic_modulus = sympify(e)
+
+    @property
+    def second_moment(self):
+        """Second moment of area of the Beam. """
+        return self._second_moment
+
+    @second_moment.setter
+    def second_moment(self, i):
+        self._second_moment = sympify(i)
+
+    @property
+    def boundary_conditions(self):
+        """
+        Returns a dictionary of boundary conditions applied on the beam.
+        """
+        return self._boundary_conditions
+
+    def apply_boundary_conditions(self, moment = [], slope = [], deflection = []):
+        """
+        Takes the boundary conditions of the beam bending problem as input.
+        The boundary conditions should be passed as keyworded arguments.
+        It is suggested to use ``moment``, ``slope`` and ``deflection`` as the
+        keywords while providing the boundary conditions of the moment curve,
+        slope curve and the deflection curve respectively as inputs.
+        A tuple in the form of (location, value) can be used to pass the
+        location and the value of a boundary condition. A list of such tuples
+        is passed under each keywords.
+
+
+        Examples
+        ========
+        >>> from sympy.physics.continuum_mechanics.beam import Beam
+        >>> from sympy import Symbol
+        >>> E = Symbol('E')
+        >>> I = Symbol('I')
+        >>> b = Beam(4, E, I)
+        >>> b.apply_boundary_conditions(moment = [(0, 4), (4, 0)], deflection = [(0, 2)], slope = [(0, 1)])
+
+        Here ``moment = [(0, 4), (4, 0)]`` means the moment due to the
+        applied loads at point ``0`` and ``4`` are `4` and `0` respectively.
+        Similarly, ``deflection = [(0, 2)]`` means the deflection of the beam
+        at point ``0`` is ``2``.
+
+        >>> b.boundary_conditions
+        {'deflection': [(0, 2)], 'moment': [(0, 4), (4, 0)], 'slope': [(0, 1)]}
+        >>> b.boundary_conditions['moment']
+        [(0, 4), (4, 0)]
```

Comment:
How does this sounds ??

Since, we are trying to construct a mutable boundary_condtions attribute, there should be an option to remove or edit the boundary conditions. So for that purpose `apply_boundary_conditions` should be used : 
```py
>>> b.apply_boundary_conditions(moment = [(0, 4), (4, 0)], deflection = [(0, 2)], slope = [(0, 1)])
>>> b.boundary_conditions
{'deflection': [(0, 2)], 'moment': [(0, 4), (4, 0)], 'slope': [(0, 1)]}
```
Now if user passes different values then this method shouldn't append them. It should change the whole list into a new list having only the passed values.
```py
>>> b.apply_boundary_conditions(moment = [(0, 4)], deflection = [(0, 4)], slope = [(0, 1), (4, 0)])
>>> b.boundary_conditions
{'deflection': [(0, 4)], 'moment': [(0, 4)], 'slope': [(0, 1), (4, 0)]}
```
Just like this.

But if the user wants to append some new boundary conditions with the existing ones, then they are free to use `apply_moment_boundary_conditions`, `apply_slope_boundary_conditions`, `apply_deflection_boundary_conditions`.

e.g.,
```py
>>> b.apply_boundary_conditions(moment = [(0, 4), (4, 0)], deflection = [(0, 2)], slope = [(0, 1)])

>>> b.boundary_conditions
{'deflection': [(0, 2)], 'moment': [(0, 4), (4, 0)], 'slope': [(0, 1)]}

>>> b.apply_moment_boundary_conditions((4, 3), (5, 0))
[(0, 4), (4, 0), (4, 3), (5, 0)]

>>> b.boundary_conditions
{'deflection': [(0, 2)], 'moment': [(0, 4), (4, 0), (4, 3), (5, 0)], 'slope': [(0, 1)]}

>>> b.apply_boundary_conditions(moment = [(0, 4)], deflection = [(0, 4)], slope = [(0, 1), (4, 0)])

>>> b.boundary_conditions
{'deflection': [(0, 4)], 'moment': [(0, 4)], 'slope': [(0, 1), (4, 0)]}
```